ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -987,7 +987,7 @@ private final class OracleQueries(
   private[http] override def containsAtContractPath(path: JsonPath, literal: JsValue) = {
     def ensureNotNull = {
       // we are only trying to reject None for an Optional record/variant/list
-      val pred: Cord = ('$' -: pathSteps(path)) ++ "?(@ != null)"
+      val pred: Cord = ('$' -: pathSteps(path)) ++ "?(!(@ == null))"
```

Comment:
Sufficient to fix the missing results with payload indexing disabled.

Alternatives that work, but also break the payload index:

```
?(exists(@) && @.type() <> "null"
```